nemo_skills/code_execution/local_sandbox/local_sandbox_server.py

```diff
@@ -117,8 +117,20 @@ def strip_ansi_codes(text):
     return output + ("\n" if output else "")
 
 
-def execute_ipython_session(generated_code, session_id, traceback_verbosity="Plain"):
-    """Execute Python code in a persistent IPython session"""
+def cleanup_session(session_id):
+    """Clean up and remove a specific session"""
+    with session_lock:
+        if session_id in sessions:
+            try:
+                del sessions[session_id]
+                logging.info(f"Cleaned up session after timeout: {session_id}")
+            except Exception as e:
```

Comment:
should we instead fail if it goes here? This would indicate something is very broken, so feels safer to just fail with error?